fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/ambassador/TransmissionSimulator.java

```diff
@@ -90,7 +90,6 @@ Map<String, TransmissionResult> preProcessInteraction(V2xMessageTransmission int
         }
 
         DestinationAddressContainer dac = interaction.getMessage().getRouting().getDestination();
-        Map<String, TransmissionResult> transmissionResults = null;
         switch (dac.getType()) {
```

Comment:
Applying it here does not improve anything.

Reply:
Could we not just return the results from the send message methods?

Reply:
Thanks to the log statements you'd need to make a block for each case. Then you need the `yield` keyword instead of `return` within the blocks. And moving the log statements into the sendMessage-methods wiould require a longer method signature. Therefore I decided to leave it here as is. We don't need to apply this new feature everywhere imho.

That would be the result:

```java
   return switch (dac.getType()) {
            case AD_HOC_TOPOCAST -> {
                if (log.isDebugEnabled()) {
                    log.debug(
                            "Send v2xMessage.id={} from node={} as Topocast (singlehop) @time={}",
                            interaction.getMessage().getId(), senderName, TIME.format(interaction.getTime())
                    );
                }
                yield sendMessageAsTopocast(senderName, dac);
            }
            case AD_HOC_GEOCAST -> {
                if (log.isDebugEnabled()) {
                    log.debug(
                            "Send v2xMessage.id={} from={} as Geocast (geo routing) @time={}",
                            interaction.getMessage().getId(), senderName, TIME.format(interaction.getTime())
                    );
                }
                yield sendMessageAsGeocast(senderName, dac);
            }
            default -> {
                log.debug(
                        "V2XMessage is not an ad hoc message. Skip this message. V2XMessage.id={}",
                        interaction.getMessage().getId()
                );
                yield null;
            }
        };
```